src/cpp/client/channel_credentials.cc

```diff
@@ -44,4 +44,29 @@ std::shared_ptr<Channel> ChannelCredentials::CreateChannelWithInterceptors(
       std::move(interceptor_creators));
 }
 
+namespace {
+class InsecureChannelCredentialsImpl final : public ChannelCredentials {
```

Comment:
I think we still want insecure creds in its own file, so that we can eventually visibility-restrict it internally.

Reply:
Alright, I've added it back.